main.go

```diff
@@ -0,0 +1,33 @@
+package main
+
+import (
+	"log"
+	"os"
+
+	"github.com/gin-gonic/gin"
+	"github.com/hoducha/ondemand-go-bootcamp/api"
+	"github.com/hoducha/ondemand-go-bootcamp/repositories"
+)
+
+func setupRouter(dataFile string) *gin.Engine {
```

Comment:
Here you are creating a complex function or a Rambo function as it is the one responsible for mapping all the layers and the only point of flexibility is the dataFile